packages/yourloops/lib/share/direct-share.api.ts

```diff
@@ -32,14 +32,26 @@ import { ShareUser } from './models/share-user.model'
 import { DirectShareAPI } from './models/direct-share-api.model'
 import { UserInvitationStatus } from '../team/models/enums/user-invitation-status.enum'
 import { UserRoles } from '../auth/models/enums/user-roles.enum'
+import HttpStatus from '../http/models/enums/http-status.enum'
+
+export const PATIENT_CANNOT_BE_ADDED_AS_CAREGIVER_ERROR_MESSAGE = 'patient-cannot-be-invited-as-caregiver'
+const PATIENT_CANNOT_BE_ADDED_AS_CAREGIVER_ERROR_CODE = HttpStatus.StatusMethodNotAllowed
 
 export default class DirectShareApi {
   static async addDirectShare(userId: string, email: string): Promise<void> {
-    await HttpService.post<void, { email: string }>({
-      url: `/confirm/send/invite/${userId}`,
-      payload: { email },
-      config: { headers: { [HttpHeaderKeys.language]: getCurrentLang() } }
-    })
+    try {
+      const { data } = await HttpService.post<void, { email: string }>({
+        url: `/confirm/send/invite/${userId}`,
+        payload: { email },
+        config: { headers: { [HttpHeaderKeys.language]: getCurrentLang() } }
+      }, [PATIENT_CANNOT_BE_ADDED_AS_CAREGIVER_ERROR_CODE])
+      return data
+    } catch (error) {
+      if (error.response.status === PATIENT_CANNOT_BE_ADDED_AS_CAREGIVER_ERROR_CODE) {
+        throw new Error(PATIENT_CANNOT_BE_ADDED_AS_CAREGIVER_ERROR_MESSAGE)
+      }
```

Comment:
Is that step necessary ? The error could be handled directly in the component when calling the API, by checking the error code. I think handling the error in the API should be done if we want to do something special when failing, in this case it just transmits the error.

Reply:
That's also what I thought at first, but it doesn't just transmit the error: it converts the data type from the API (containing `response.status`) into a basic `Error` type. 
We thought it was best for the components to always handle `Error` objects (which only contain (`message`, `name`, `cause`, `stack`) rather than having to make some distinctions and in some rare cases, handle a different type of data for the error thrown. So the API has the responsibility to convert any type of error thrown by the backend into an `Error` object, and the components only handle `Error` objects